src/main.rs

```diff
@@ -393,7 +380,11 @@ fn main() {
         if let Some(lookup_result) = cache.find_page(
             &command,
             &languages,
-            config.directories.custom_pages_dir.as_deref(),
+            config
+                .directories
+                .custom_pages_dir
+                .as_ref()
+                .map(|path_with_source| path_with_source.path.as_ref()),
```

Comment:
This looks a bit odd, it can be more concise if we implement a `&Path` getter on `PathWithSource`:
```suggestion
            config
                .directories
                .custom_pages_dir
                .as_ref()
                .map(PathWithSource::path),
```
(same 20 lines above)